ex02.exs

```diff
@@ -1,6 +1,25 @@
 
 defmodule Ex02 do
 
+  def new_counter(value) do
+    {:ok, count} = Agent.start(fn -> value - 1 end)
+    count
+  end
+
+  def next_value(count) do
+    Agent.update(count, &(&1 + 1))
```

Comment:
## -4

This is actually a bug—there's a race condition. Two different processes could call this function at the same time. The first does the update, and then the scheduler suspends it and runs the second. It also does the update and then gets the updated value. The first process then gets scheduled and gets the value, but by this time it has been incremented twice.

This is why there's `Agent.get_and_update`